fixieai/client/agent.py

```diff
@@ -18,44 +18,50 @@ class Agent:
 
     Args:
         client: The FixieClient instance to use.
-        handle: The handle of the Agent.
+        agent_id: The Agent ID, e.g., "fixie/calc", or handle, e.g., "dice".
     """
 
     def __init__(
         self,
         client: FixieClient,
-        handle: str,
+        agent_id: str,
     ):
         self._client = client
         self._gqlclient = self._client.gqlclient
-        self._handle = handle
+
+        self._owner: Optional[str] = None
+        if "/" in agent_id:
+            self._owner, self._handle = agent_id.split("/")
+        else:
+            self._handle = agent_id
+
         self._metadata: Optional[Dict[str, Any]] = None
         try:
             self._metadata = self.get_metadata()
         except:
             self._metadata = None
 
-    @property
-    def handle(self) -> Optional[str]:
-        """Return the handle for this Agent."""
-        return self._handle
-
     @property
     def agent_id(self) -> Optional[str]:
         """Return the agentId for this Agent."""
         if self._metadata is None:
             return None
         agent_id = self._metadata["agentId"]
-        assert isinstance(agent_id, str)
+        assert agent_id is None or isinstance(agent_id, str)
```

Comment:
when would this be None? same for the fields below, shouldn't they be "" at worst?

Reply:
Because the agent might not exist yet